scipy/linalg/_linalg_inv.c

```diff
@@ -0,0 +1,277 @@
+                        data[i * n + j] = NAN;
+                    }
+                }
+            } else {
+                // If singular, set isIllconditioned to 1
+                *isIllconditioned = 1;
+            }
+        }
+        // Swap back to original order
+        swap_cf_s(data, &ret_data[idx*n*n], n, n, n);
+    }
+
+free_exit:
+    free(buffer);
+    free(ipiv);
+    return;
+}
+
+
+void _inverse_d(const PyArrayObject* ap_Am, double* restrict ret_data, int* isIllconditioned, int* isSingular, int* info)
+{
+    *isIllconditioned = 0;
+    *isSingular = 0;
+    int lower_band = 0, upper_band = 0;
+
+    // --------------------------------------------------------------------
+    // Input Array Attributes
+    // --------------------------------------------------------------------
+    double* restrict Am_data = (double*)PyArray_DATA(ap_Am);
+    int ndim = PyArray_NDIM(ap_Am);              // Number of dimensions
+    npy_intp* shape = PyArray_SHAPE(ap_Am);      // Array shape
+    npy_intp n = shape[ndim - 1];                // Slice size
+    npy_intp* restrict strides = PyArray_STRIDES(ap_Am);
+    // Get the number of slices to traverse if more than one; np.prod(shape[:-2])
+    npy_intp outer_size = 1;
+    if (ndim > 2)
+    {
+        for (int i = 0; i < ndim - 2; i++) { outer_size *= shape[i];}
+    }
+
+    // --------------------------------------------------------------------
+    // Workspace computation and allocation
+    // --------------------------------------------------------------------
+    double tmp_float = 0.0;
+    int intn = (int)n, lwork = -1;
+    dgetri_(&intn, NULL, NULL, NULL, &tmp_float, &lwork, &info);
+    if (info != 0) { *info = -100; return; }
+    lwork = (int)tmp_float;
+    lwork = (4*n > lwork ? 4*n : lwork); // gecon needs at least 4*n
+    double* buffer = malloc((n*n + lwork)*sizeof(double));
+    if (NULL == buffer) { *info = -101; return; }
+    // Chop buffer into two parts, one for data and one for work
+    double* restrict data = &buffer[0];
+    double* restrict work = &buffer[n*n];
+    int* ipiv = malloc(n*sizeof(int));
+    if (ipiv == NULL) { free(buffer); *info = -102; return; }
+
+    // Main loop to traverse the slices
+    for (npy_intp idx = 0; idx < outer_size; idx++) {
+
+        npy_intp offset = 0;
+        npy_intp temp_idx = idx;
+        for (int i = ndim - 3; i >= 0; i--) {
+            offset += (temp_idx % shape[i]) * strides[i];
+            temp_idx /= shape[i];
+        }
+        double* restrict slice_ptr = (double*)(Am_data + (offset/sizeof(double)));
+
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < n; j++) {
+                work[i * n + j] = *(slice_ptr + (i*strides[ndim - 2]/sizeof(double)) + (j*strides[ndim - 1]/sizeof(double)));
+            }
+        }
+        swap_cf_d(work, data, n, n, n);
+
+        // Check if array is triangular
+        bandwidth_s(data, n, n, &lower_band, &upper_band);
```

Comment:
```suggestion
        bandwidth_d(data, n, n, &lower_band, &upper_band);
```

To double-check: this is a typo, right? Otherwise `bandwidth_s` expects a `float*` but is called with `double*` argument

Reply:
Yes that's right. I've just noticed another one in `bandwidth_c,z` functions, the zero checking condition should be `||` instead of `&&`.